projects/elements-demo/src/app/features/examples/basic/basic.component.html

```diff
@@ -114,11 +114,7 @@ <h2>Advanced binding (objects instead of primitives)</h2>
       <ng-template #loading>Loading...</ng-template>
       <ng-template #error>Loading failed...</ng-template>
       <ion-item
-        *axLazyElement="
-          'https://unpkg.com/@ionic/core@4.6.2/dist/ionic/ionic.js';
-          loadingTemplate: loading;
-          errorTemplate: error
-        "
+        *axLazyElement="null; loadingTemplate: loading; errorTemplate: error"
```

Comment:
If null is necessary that pretty bad, we have to come up with a way to use it without null... This is not a nice syntax...

Reply:
I mean we could introduce another directive, like `axLazyElementWithConfig` that way we can have other API without breaking change, also I like the config for the loading in the `forRoot`, how would that work with components ? as original thing was with templates, doesn't component complicate things a lot in terms of creation and that they might have their own dependencies etc...  ?

Reply:
We are just adding a feature @tomastrajan, the existing directive will work as-is, plus we will add additional features using  `forRoot`. Rendering a component or template dynamically is not complicated at all in Angular, we have to only add one additional dependency of ComponentResolverFactory to our directive. and use the following code to render.

```ts
const config = elementsLoaderService.getElementConfig(elementTag);
if (this.loadingTemplateRef) {
      this.vcr.createEmbeddedView(this.loadingTemplateRef);
} else if ((config && config.loadingComponent) || options.loadingComponent) {
      const factory = this.componentResolverFactory. resolveComponentFactory(config.loadingComponent || options.loadingComponent);
      this.vcr.createComponent(factory);
}
```

In the future, if you want to add configs using the service from anywhere in the code (which we should be able to do after this code merge too), and want to pass template instead of component, maybe we can have `loadingTemplate` added too in the `ElementConfig`,

```ts
this.elementsLoaderService.addConfig({
    tag: 'ce-element',
    url,
    loadingTemplate: this.templateRef,
});
```